Prox/Prox/Utilities/CategoriesUtil.swift

```diff
@@ -124,6 +124,33 @@ struct CategoriesUtil {
         }
         return categoryToName
     }()
+
+    static let categoryToFilter: [String: Filter] = {
+        var categoryToFilter = [String: Filter]()
+
+        // Build the initial map of categories -> filters that we set explicitly.
+        for (filter, categories) in Filter.categories {
+            for category in categories {
+                categoryToFilter[category] = filter
+            }
+        }
+
+        // Set the remaining categories based on the first ancestor with an explicitly-set filter.
+        for category in categoryToParentsMap.keys {
+            guard let firstCategorizedParent = getAncestors(forCategory: category).flatMap({ categoryToFilter[$0] }).first else { continue }
+            categoryToFilter[category] = firstCategorizedParent
+        }
+
+        return categoryToFilter
+    }()
+
+    /// Returns this category, its root categories, and all parents in between.
+    /// The result is DFS-ordered.
```

Comment:
The results is DFS-ordered but our Yelp categories are not guaranteed to be returned in any order, right? afaik, Yelp doesn't put the parents array in any particular order and could resort them when the file is updated. This means our behavior could change if we add a new Yelp categories list.

I think we should add a sort in our `categoryToParentsMap` init code.